src/Month.js

```diff
@@ -60,8 +54,14 @@ class MonthView extends React.Component {
     )
   }
 
-  componentDidUpdate() {
-    if (this.state.needLimitMeasure) this.measureRowLimit(this.props)
+  componentDidUpdate(prevProps) {
+    if (!dates.eq(prevProps.date, this.props.date, 'month')) {
```

Comment:
this sets should be collapsed and the state removed. this check just determines if the measureRowLimit should run, so instead of setting state, do the work instead